bliss/models/galaxy_net.py

```diff
@@ -197,26 +288,210 @@ def plot_reconstruction(
+                ax_recon_res.set_title("Initial Residual Reconstruction", pad=pad)
 
             # standarize ranges of true and reconstruction
             image = images[idx, 0].detach().cpu().numpy()
             recon = recon_mean[idx, 0].detach().cpu().numpy()
             residual = residuals[idx, 0].detach().cpu().numpy()
+            recon_mean_main_i = recon_mean_main[idx, 0].detach().cpu().numpy()
+            residuals_main_i = residuals_main[idx, 0].detach().cpu().numpy()
+            recon_mean_residual_i = recon_mean_residual[idx, 0].detach().cpu().numpy()
             vmin = min(image.min().item(), recon.min().item())
             vmax = max(image.max().item(), recon.max().item())
 
             # plot images
             plot_image(fig, ax_true, image, vrange=(vmin, vmax))
             plot_image(fig, ax_recon, recon, vrange=(vmin, vmax))
             plot_image(fig, ax_res, residual, vrange=(vmin_res, vmax_res))
+            plot_image(fig, ax_recon_main, recon_mean_main_i, vrange=(vmin, vmax))
+            plot_image(fig, ax_res_main, residuals_main_i, vrange=(vmin_res, vmax_res))
+            plot_image(fig, ax_recon_res, recon_mean_residual_i, vrange=(vmin_res, vmax_res))
 
         plt.tight_layout()
-
         return fig
 
-    def test_step(self, batch, batch_idx):
-        """Testing step (pytorch lightning)."""
-        images, background = batch["images"], batch["background"]
-        recon_mean = self(images, background)
+    def _plot_grid_examples(self, outputs):
+        images, recon_mean = outputs["images"], outputs["recon_mean"]
+        # 1.  plot a grid of all input images and recon_mean
+        # 2.  only plot the highest band
+
+        nrow = 16
         residuals = (images - recon_mean) / torch.sqrt(images)
-        self.log("max_residual", residuals.abs().max())
+
+        image_grid = make_grid(images, nrow=nrow)[0]
+        recon_grid = make_grid(recon_mean, nrow=nrow)[0]
+        residual_grid = make_grid(residuals, nrow=nrow)[0]
+        h, w = image_grid.size()
+        base_size = 8
+        fig = plt.figure(figsize=(3 * base_size, int(h / w * base_size)))
+        for i, grid in enumerate([image_grid, recon_grid, residual_grid]):
+            plt.subplot(1, 3, i + 1)
+            plt.imshow(grid.cpu().numpy(), interpolation=None)
+            if i == 0:
+                plt.title("images")
+            elif i == 1:
+                plt.title("recon_mean")
+            else:
+                plt.title("residuals")
+            plt.xticks([])
+            plt.yticks([])
+        return fig
+
+
+class OneCenteredGalaxyEncoder(nn.Module):
+    def __init__(self, main_encoder, main_decoder, residual_encoder, slen=None, allow_pad=False):
+        super().__init__()
+        self.main_encoder = main_encoder
+        self.main_decoder = main_decoder
+        self.residual_encoder = residual_encoder
+        self.slen = slen
+        self.allow_pad = allow_pad
```

Comment:
Could you explain why you have this padding? or what exactly does `allow_pad = True` does?

Reply:
The reason I add padding is because the galaxy encoder passes images of size 44x44, while the encoder needs an image of size 53x53. 

I'll write a docstring for this class so this reason is documented in the code